.github/workflows/test.yml

```diff
@@ -33,0 +1,32 @@
+name: Test
+
+on:
+  pull_request:
+    branches:
+      - "master"
+
+jobs:
+  test:
+    runs-on: ubuntu-latest
+    timeout-minutes: 10
+
+    steps:
+      - name: Checkout repo
+        uses: actions/checkout@v2
+
+      - name: Set Up Python 3.10
+        uses: actions/setup-python@v2
+        with:
+          python-version: "3.10"
+
+      - name: Install requirements.txt
+        run: |
+          python -m pip install --upgrade pip
+          pip install -r requirements.txt
+
+      - name: Run flake8
+        run: flake8
+
+      - name: Run tests
+        timeout-minutes: 5

```

Comment:
The `timeout-minutes` parameter is redundantly specified here. It is already set at the job level (line 11). You can remove this line to avoid redundancy.